libbeat/common/useragent/useragent.go

```diff
@@ -20,15 +20,29 @@ package useragent
 import (
 	"fmt"
 	"runtime"
+	"strings"
 
 	"github.com/elastic/beats/v7/libbeat/version"
 )
 
 // UserAgent takes the capitalized name of the current beat and returns
 // an RFC compliant user agent string for that beat.
-func UserAgent(beatNameCapitalized string) string {
-	return fmt.Sprintf("Elastic-%s/%s (%s; %s; %s; %s)",
-		beatNameCapitalized,
-		version.GetDefaultVersion(), runtime.GOOS, runtime.GOARCH,
-		version.Commit(), version.BuildTime())
+func UserAgent(beatNameCapitalized string, additional_agents ...string) string {
+	var builder strings.Builder
+	fmt.Fprintf(&builder, "Elastic-%s/%s ", beatNameCapitalized, version.GetDefaultVersion())
+	uaValues := []string{
+		runtime.GOOS,
+		runtime.GOARCH,
+		version.Commit(),
+		version.BuildTime().String(),
+	}
+	if additional_agents != nil {
+		for _, val := range additional_agents {
+			if val != "" {
+				uaValues = append(uaValues, val)
+			}
+		}
+	}
+	fmt.Fprintf(&builder, "(%s)", strings.Join(uaValues, "; "))
```

Comment:
When using a strings.Builder it's best to avoid fprintf from an allocation perspective. Something along the lines of this is optimal...

```
builder.WriteByte('(')
builder.WriteString(strings.Join())
builder.WriteByte(')')
```